python/lsst/pipe/tasks/multiBand.py

```diff
@@ -763,33 +780,68 @@ def mergeCatalogs(self, catalogs, patchRef):
         # This first zip iterates over all the catalogs simultaneously, yielding a sequence of one
         # record for each band, in order.
         for n, orderedRecords in enumerate(zip(*orderedCatalogs)):
-            # Now we iterate over those record-band pairs, until we find the one with the right flag set.
+            # Now we iterate over those record-band pairs, keeping track of the priority and largest S/N band
```

Comment:
I think a few more comments scattered through this code would be useful: while there's nothing desperately complicated going on, it took me a few moments of head-scratching to grok the logic.